File: app/src/main/java/com/wayfair/themoviedb/data/model/response/Result.java

package com.wayfair.themoviedb.data.model.response;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

public class Result {

    @SerializedName("original_name")
    @Expose
    public String originalName;
    @SerializedName("id")
    @Expose
    public Integer id;
    @SerializedName("media_type")
    @Expose
    public String mediaType;
    @SerializedName("name")
    @Expose
    public String name;
    @SerializedName("vote_count")
    @Expose
    public Integer voteCount;
    @SerializedName("vote_average")
    @Expose
    public Double voteAverage;
    @SerializedName("poster_path")
    @Expose
    public String posterPath;
    @SerializedName("first_air_date")
    @Expose
    public String firstAirDate;
    @SerializedName("popularity")
    @Expose
    public Double popularity;
    @SerializedName("genre_ids")
    @Expose
    public List<Integer> genreIds = null;
    @SerializedName("original_language")
    @Expose
    public String originalLanguage;
    @SerializedName("backdrop_path")
    @Expose
    public Object backdropPath;
    @SerializedName("overview")
    @Expose
    public String overview;
    @SerializedName("origin_country")
    @Expose
    public List<String> originCountry = null;
    @SerializedName("video")
    @Expose
    public Boolean video;
    @SerializedName("title")
    @Expose
    public String title;
    @SerializedName("original_title")
    @Expose
    public String originalTitle;
    @SerializedName("adult")
    @Expose
    public Boolean adult;
    @SerializedName("release_date")
    @Expose
    public String releaseDate;
    @SerializedName("profile_path")
    @Expose
    public Object profilePath;
    @SerializedName("known_for")
    @Expose
    public List<KnownFor> knownFor = null;

}

File: app/src/main/java/com/wayfair/themoviedb/data/model/response/Result.kt
package com.wayfair.themoviedb.data.model.response

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class Result {

    @SerializedName("original_name")
    @Expose
    var originalName: String? = null
    @SerializedName("id")
    @Expose
    var id: Long? = null
    @SerializedName("media_type")
    @Expose
    var mediaType: String? = null
    @SerializedName("name")
    @Expose
    var name: String? = null
    @SerializedName("vote_count")
    @Expose
    var voteCount: Long? = null
    @SerializedName("vote_average")
    @Expose
    var voteAverage: Float? = null
    @SerializedName("poster_path")
    @Expose
    var posterPath: String? = null
    @SerializedName("profile_path")
    @Expose
    var profilePath: String? = null
    @SerializedName("first_air_date")
    @Expose
    var firstAirDate: String? = null
    @SerializedName("popularity")
    @Expose
    var popularity: Double? = null
    @SerializedName("genre_ids")
    @Expose
    var genreIds: List<Int>? = null
    @SerializedName("original_language")
    @Expose
    var originalLanguage: String? = null
    @SerializedName("backdrop_path")
    @Expose
    var backdropPath: Any? = null
    @SerializedName("overview")
    @Expose
    var overview: String? = null
    @SerializedName("origin_country")
    @Expose
    var originCountry: List<String>? = null
    @SerializedName("video")
    @Expose
    var video: Boolean? = null
    @SerializedName("title")
    @Expose
    var title: String? = null
    @SerializedName("original_title")
    @Expose
    var originalTitle: String? = null
    @SerializedName("adult")
    @Expose
    var adult: Boolean? = null
    @SerializedName("release_date")
    @Expose
    var releaseDate: String? = null
    @SerializedName("known_for")
    @Expose
    var knownFor: List<KnownFor>? = null

}
